public/src/components/dashboard/SubmissionsTable.tsx

```diff
@@ -38,15 +38,15 @@ const SubmissionsTable = (): JSX.Element => {
             }
             // Submissions are indexed by the assignment order - 1.
             const submission = submissions[assignment.order - 1] ?? new Submission()
-            if (submission.status !== Submission_Status.APPROVED) {
+            if (submission.status !== Submission_Status.APPROVED && assignment.deadline) {
                 const deadline = timeFormatter(assignment.deadline)
                 if (deadline.daysUntil > 3 && submission.score >= assignment.scoreLimit) {
                     deadline.className = "table-success"
                 }
                 if (!deadline.message) {
```

Comment:
We should always set `deadline.message` to a valid string, e.g., `--` if there is no message (see Line 61).